backend/apps/content/bigo_complexity_challenge_plugin.py

```
@@ -30,24 +30,45 @@
+            "ValueError": ValueError, "TypeError": TypeError,
+        },
+        "__random": random,
+    }
+
+    try:
+        # Pre-compile and execute the setup + function definition
+        exec(full_code, safe_globals)
+        
+        # Now measure execution time of just the function call
+        start_time = time.time()
+        exec(run_code, safe_globals)
```

Ruff (0.16.0)

[error] Use of `exec` detected

(S102)